File: 2023/01/One.java
import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Path;
import java.util.stream.Stream;

void main() throws IOException {
    try (final Stream<String> lines = Files.lines(Path.of("input.txt"))) {
        final int sum = lines.map(this::prepareLine)
                .mapToInt(this::findCalibrationValue)
                .sum();
        System.out.println(sum);
    }
}

private String prepareLine(final String line) {
    return line
            .replaceAll("one", "o1e")
            .replaceAll("two", "t2o")
            .replaceAll("three", "t3e")
            .replaceAll("four", "4")
            .replaceAll("five", "5e")
            .replaceAll("six", "6")
            .replaceAll("seven", "7n")
            .replaceAll("eight", "8t")
            .replaceAll("nine", "n9e");
}

private int findCalibrationValue(final String line) {
    final int[] digits = line.chars()
            .filter(Character::isDigit)
            .map(Character::getNumericValue)
            .toArray();
    final int calibrationValue = digits[0] * 10 + digits[digits.length - 1];
    System.out.println(line + " -> " + calibrationValue);
    return calibrationValue;
}

File: 2023/01/one.rs
fn main() {
    let input = include_str!("input.txt");
    let sum = input.split("\n")
        .map(prepare_line)
        .map(find_calibration_value)
        .sum::<u32>();
    println!("{sum}");
}

fn prepare_line(line: &str) -> String {
    line.replace("one", "o1e")
        .replace("two", "t2o")
        .replace("three", "t3e")
        .replace("four", "4")
        .replace("five", "5e")
        .replace("six", "6")
        .replace("seven", "7n")
        .replace("eight", "8t")
        .replace("nine", "n9e")
}

fn find_calibration_value(line: String) -> u32 {
    let digits = line.chars()
        .filter(|c| c.is_ascii_digit())
        .map(|c| c.to_digit(10).unwrap())
        .collect::<Vec<u32>>();
    let calibration_value = digits[0] * 10 + digits.last().unwrap();
    println!("{line} -> {calibration_value}");
    calibration_value
}